types/block.go

```diff
@@ -259,6 +264,107 @@ func mustPush(rowTree *nmt.NamespacedMerkleTree, id namespace.ID, data []byte) {
 	}
 }
 
+// PutBlock posts and pins erasured block data to IPFS using the provided
+// ipld.NodeAdder. Note: the erasured data is currently recomputed
+func (b *Block) PutBlock(ctx context.Context, nodeAdder format.NodeAdder) error {
+	if nodeAdder == nil {
+		return errors.New("no ipfs node adder provided")
+	}
+
+	// recompute the shares
+	namespacedShares := b.Data.computeShares()
+	shares := namespacedShares.RawShares()
+
+	// don't do anything if there is no data to put on IPFS
+	if len(shares) == 0 {
+		return nil
+	}
+
+	// recompute the eds
+	eds, err := rsmt2d.ComputeExtendedDataSquare(shares, rsmt2d.RSGF8, rsmt2d.NewDefaultTree)
+	if err != nil {
+		return fmt.Errorf("failure to recompute the extended data square: %w", err)
+	}
+
+	// add namespaces to erasured shares and flatten the eds
+	leaves := flattenNamespacedEDS(namespacedShares, eds)
+
+	// iterate through each set of col and row leaves
+	for _, leafSet := range leaves {
+		// create a batch per each leafSet
+		batchAdder := nodes.NewNmtNodeAdder(ctx, format.NewBatch(ctx, nodeAdder))
```

Comment:
I think it is better to extract a local var from `format.NewBatch(ctx, nodeAdder)` and use that to commit to that batch.